shelter/remoteattestation/wolfssl-ra-challenger.c

```diff
@@ -0,0 +1,360 @@
+/**
+ * wolfSSL-based implementation of the RA-TLS challenger API
+ * (cf. ra-challenger.h).
+ */
+
+
+#include <assert.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <wolfssl/options.h>
+#include <wolfssl/ssl.h>
+#include <wolfssl/wolfcrypt/asn.h>
+#include <wolfssl/wolfcrypt/asn_public.h>
+#include <wolfssl/wolfcrypt/coding.h>
+#include <wolfssl/wolfcrypt/rsa.h>
+#include <wolfssl/wolfcrypt/sha256.h>
+#include <wolfssl/wolfcrypt/signature.h>
+
+#ifdef RATLS_ECDSA
+#include <SgxEcdsaAttestation/QuoteVerification.h>
+#endif
+
+#include "ra.h"
+#include "wolfssl-ra.h"
+#include "ra-challenger.h"
+#include "ra-challenger_private.h"
+
+#ifdef RATLS_ECDSA
+#include "ecdsa-sample-data/real/sample_data.h"
+#endif
+
+extern unsigned char ias_sign_ca_cert_der[];
+extern unsigned int ias_sign_ca_cert_der_len;
+
+void get_quote_from_cert
```

Comment:
Is it possible to refer to sgx-tools/get-ias-report.go to implement the logic of generating ias report? Only tls part refers to wolfssl.

Reply:
Per mu understanding, this task will for the next stage of shelter get ias report by shelter instead of by inclavared?

Reply:
I mean the back port codes include the logics of ias report generation and tls related. Actually we hate using wolfssl written in C inside golang context (shelter is a golang app right?). So if we can re-use existing golang codes, it is better.

Reply:
Regarding of quote-only generation, this is another story. Of course it is a cause point to support my point. But we need to implement quote-only generation in inclavared side. This work will be done with dcap soon but not soon for epic.